src/scripts/main.js

```diff
@@ -1,3 +1,265 @@
+  const rowsForSorting = [...tbody.children];
+
+  if (!titleReverse || !thead.contains(titleReverse)) {
+    return;
+  }
+
+  switch (titleReverse.innerText) {
+    case 'Name':
+    case 'Position':
+    case 'Office':
+      rowsForSorting.sort((current, next) => {
+        const currentCellString = current.cells[titleIndexReverse].innerText;
+        const nextCellString = next.cells[titleIndexReverse].innerText;
+
+        return nextCellString.localeCompare(currentCellString);
+      });
+      break;
+
+    case 'Age':
+    case 'Salary':
+      rowsForSorting.sort((current, next) => {
+        const currentCellNum = current.cells[titleIndexReverse].innerText;
+        const nextCellNum = next.cells[titleIndexReverse].innerText;
+        const convertedCurrentNum = converter(currentCellNum);
+        const convertedNextNum = converter(nextCellNum);
+
+        return convertedNextNum - convertedCurrentNum;
+      });
+      break;
+  }
+
+  rows.forEach(row => tbody.removeChild(row));
+
+  rowsForSorting.forEach(newRow => tbody.appendChild(newRow));
+}
+
+const form = document.createElement('form');
+
+form.classList.add('new-employee-form');
+form.method = 'GET';
+form.action = '#';
+
+form.insertAdjacentHTML('afterbegin', `
+<label>Name:
+  <input
+    class="cell-input"
+    type="text"
+    name="name"
+    data-qa="name"
+    required
+  >
+</label>
+<label>Position:
+  <input
+    class="cell-input"
+    type="text"
+    name="position"
+    data-qa="position"
+    required
+  >
+</label>
+<label>Office:
+  <select
+    class="cell-input"
+    name="office"
+    data-qa="office"
+    required
+  >
+    <option value="Tokyo">Tokyo</option>
+    <option value="Singapore">Singapore</option>
+    <option value="London">London</option>
+    <option value="New York">New York</option>
+    <option value="Edinburgh">Edinburgh</option>
+    <option value="San Francisco">San Francisco</option>
+  </select>
+</label>
+<label>Age:
+  <input
+    class="cell-input"
+    type="number"
+    name="age"
+    data-qa="age"
+    required
+  >
+</label>
+<label>Salary:
+  <input
+    class="cell-input"
+    type="number"
+    name="salary"
+    data-qa="salary"
+    required
+  >
+</label>
+<button type="submit">Save to table</button>
+  `);
+
+document.body.append(form);
+
+const buttonForSaving = document.querySelector('button');
+const input = document.querySelectorAll('input');
+const select = document.querySelector('select');
+
+buttonForSaving.addEventListener('click', saveToTable);
+
+function conditionChecker(nameInput, position, age) {
+  if (nameInput.length < 4) {
+    pushNotification(
+      150, 10,
+      'Error invalid input',
+      'The length of the name must be longer',
+      'error'
+    );
+  }
+
+  if (!position.length) {
+    pushNotification(
+      10, 10,
+      'Error invalid input',
+      'Position input required',
+      'error'
+    );
+  }
+
+  if (age < 18 || age > 90) {
```

Comment:
```suggestion
  if (nameInput.length < 4) {
    pushNotification(
      150, 10,
      'Error invalid input',
      'The length of the name must be longer',
      'error'
    );
  }

  if (!position.length) {
    pushNotification(
      10, 10,
      'Error invalid input',
      'Position input required',
      'error'
    );
  }

  if (age < 18 || age > 90) {
```
что такое 18, 90, 4?